tracker/progress.go

```diff
@@ -115,17 +123,17 @@ func (pr *Progress) BecomeProbe() {
 	if pr.State == StateSnapshot {
 		pendingSnapshot := pr.PendingSnapshot
 		pr.ResetState(StateProbe)
-		pr.Next = max(pr.Match+1, pendingSnapshot+1)
+		pr.Next = max(pr.Match+1, pendingSnapshot+1) // invariant: Match < Next
```

Comment:
I see there are comments "`// invariant: Match < Next`" in multiple places. I think we only need to keep the comment in `type Progress struct`, and also add verification (I was planning to do it for a while) to safeguard it instead of adding comment everywhere.

Reply:
Removed the comment from all the places, except the non-trivial `max(c.LastIndex, 1)` in one place (where this invariant was not maintained before this PR).